### PR TITLE
fix(anta.tests): Cleaning up AVT tests module (VerifyAVTSpecificPath)

### PR DESCRIPTION
**Pull request by vitthalmagadum**
# Description

Refactoring AVT tests module to address the following issues:

- https://github.com/aristanetworks/anta/issues/809
- https://github.com/aristanetworks/anta/issues/587
- https://github.com/aristanetworks/anta/issues/388

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have run pre-commit for code linting and typing (`pre-commit run`)
- [ ] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes (`tox -e testenv`)

### CODE REVIEW

File: anta/input_models/avt.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+        Examples
+        --------
+        AVT CONTROL-PLANE-PROFILE VRF: default (Destination: 10.101.255.2, Next-hop: 10.101.255.1)
+
+        """
+        return f"AVT {self.avt_name} VRF: {self.vrf} (Destination: {self.destination}, Next-hop: {self.next_hop})"
+
+
+class AVTPaths(AVTPath):  # pragma: no cover
```
Comment (MaheshGSLAB) [inline anta/input_models/avt.py:41]: this is not testcase name so dont know why we are deprecating this class name. We can change it directly.

Comment (MaheshGSLAB) [reply anta/input_models/avt.py:41]: @carl-baillargeon @gmuloc

Comment (gmuloc) [reply anta/input_models/avt.py:41]: yeah we don;t need this one - we only need to import the ClassVar properly using the new model

File: anta/tests/avt.py [hunk 2]
```diff
@@ -94,35 +100,16 @@ class VerifyAVTSpecificPath(AntaTest):
...
-            """The IPv4 address of the AVT peer."""
-            next_hop: IPv4Address
-            """The IPv4 address of the next hop for the AVT peer."""
-            path_type: str | None = None
-            """The type of the AVT path. If not provided, both 'direct' and 'multihop' paths are considered."""
-
-    def render(self, template: AntaTemplate) -> list[AntaCommand]:
-        """Render the template for each input AVT path/peer."""
-        return [template.render(vrf=path.vrf, avt_name=path.avt_name, destination=path.destination) for path in self.inputs.avt_paths]
+        AVTPaths: ClassVar[type[AVTPaths]] = AVTPaths
```
Comment (gmuloc) [inline anta/tests/avt.py:111]:
```suggestion
        AVTPaths: ClassVar[type[AVTPath]] = AVTPath
```

File: anta/tests/avt.py [hunk 3]
```diff
@@ -94,35 +100,16 @@ class VerifyAVTSpecificPath(AntaTest):
     ```
     """
 
-    description = "Verifies the status and type of an AVT path for a specified VRF."
     categories: ClassVar[list[str]] = ["avt"]
-    commands: ClassVar[list[AntaCommand | AntaTemplate]] = [
-        AntaTemplate(template="show adaptive-virtual-topology path vrf {vrf} avt {avt_name} destination {destination}")
-    ]
+    commands: ClassVar[list[AntaCommand | AntaTemplate]] = [AntaCommand(command="show adaptive-virtual-topology path", revision=1)]
```
[the remark this answers is not in the document]
Comment (gmuloc) [reply anta/tests/avt.py:104]: good point - any reason we are changing this? (not sure I was involved int he discussion)

[the remark this answers is not in the document]
Comment (vitthalmagadum) [reply anta/tests/avt.py:104]: understand the concern about the large output of the show adaptive-virtual-topology path command. The reason for switching to this command is to utilize cached output, which improves efficiency.
We can have something later on for the reporting IMO.

[the remark this answers is not in the document]
Comment (vitthalmagadum) [reply anta/tests/avt.py:104]: Also my concern is about the number of lines of the code and the rendering of multiple commands.

[the remark this answers is not in the document]
Comment (vitthalmagadum) [reply anta/tests/avt.py:104]: @gmuloc @carl-baillargeon @MaheshGSLAB 
Could you please provide guidance on how we should proceed with this?

[the remark this answers is not in the document]
Comment (carl-baillargeon) [reply anta/tests/avt.py:104]: Moving forward we should use AntaCommand vs AntaTemplate. 1 command = 1 request so it's more efficient, especially at high scale. The drawback is it's using more memory but it shouldn't be a problem. If it becomes a problem we can add a knob like we did with `VerifyRoutingTableEntries` to control how we collect data for this test.

[the remark this answers is not in the document]
Comment (vitthalmagadum) [reply anta/tests/avt.py:104]: Sure. Thanks @carl-baillargeon

File: anta/tests/avt.py [hunk 4]
```diff
@@ -131,59 +118,38 @@ def test(self) -> None:
...
-            peer = str(command.params.destination)
-
-            command_output = command.json_output.get("vrfs", {})
-
-            # If no AVT is configured, mark the test as failed and skip to the next command
-            if not command_output:
-                self.result.is_failure(f"AVT configuration for peer '{peer}' under topology '{avt_name}' in VRF '{vrf}' is not found.")
-                continue
+        for avt_path in self.inputs.avt_paths:
+            if (path_output := get_value(self.instance_commands[0].json_output, f"vrfs.{avt_path.vrf}.avts.{avt_path.avt_name}.avtPaths")) is None:
```
[the remark this answers is not in the document]
Comment (vitthalmagadum) [reply anta/tests/avt.py]: Updated the check.
Thanks!!

File: anta/tests/avt.py (hunk 2 above)
Comment (vitthalmagadum) [reply anta/tests/avt.py]: Updated

File: anta/input_models/avt.py (hunk 1 above)
Comment (vitthalmagadum) [reply anta/input_models/avt.py]: Added classVar. Thanks!!

**Review by gmuloc — APPROVED** [review]

**Pull request merged**